pkg/driver/docker/docker.go

```diff
@@ -363,6 +366,188 @@ func (d *dockerDriver) RunDockerDevContainer(
 		return err
 	}
 
+	if runtime.GOOS != "windows" && ((parsedConfig.ContainerUser != "" || parsedConfig.RemoteUser != "") &&
+		(parsedConfig.UpdateRemoteUserUID == nil || *parsedConfig.UpdateRemoteUserUID)) {
+		// Retrieve local user UID and GID
+		localUser, err := user.Current()
+		if err != nil {
+			return err
+		}
+		localUid := localUser.Uid
+		localGid := localUser.Gid
+
+		// Retrieve user to update
+		containerUser := parsedConfig.RemoteUser
+		if containerUser == "" {
+			containerUser = parsedConfig.ContainerUser
+		}
+		container, err := d.FindDevContainer(ctx, workspaceId)
+		if err != nil {
+			return err
+		} else if container == nil {
+			return nil
+		}
+
+		// Create temporary files to store /etc/passwd and /etc/group
+		containerPasswdFileIn, err := os.CreateTemp("", "devpod_container_passwd_in")
+		if err != nil {
+			return err
+		}
+		containerGroupFileIn, err := os.CreateTemp("", "devpod_container_group_in")
+		if err != nil {
+			return err
+		}
+		containerPasswdFileOut, err := os.CreateTemp("", "devpod_container_passwd_out")
+		if err != nil {
+			return err
+		}
+		containerGroupFileOut, err := os.CreateTemp("", "devpod_container_group_out")
+		if err != nil {
+			return err
+		}
+		defer os.Remove(containerPasswdFileIn.Name())
+		defer os.Remove(containerGroupFileIn.Name())
+		defer os.Remove(containerPasswdFileOut.Name())
+		defer os.Remove(containerGroupFileOut.Name())
+
+		// Copy /etc/passwd and /etc/group from the container to the temporary files
+		{
+			args := []string{"cp", fmt.Sprintf("%s:/etc/passwd", container.ID), containerPasswdFileIn.Name()}
+			d.Log.Debugf("Running docker command: %s %s", d.Docker.DockerCommand, strings.Join(args, " "))
+			err = d.Docker.Run(ctx, args, nil, writer, writer)
+			if err != nil {
+				return err
+			}
+		}
+
+		{
+			args := []string{"cp", fmt.Sprintf("%s:/etc/group", container.ID), containerGroupFileIn.Name()}
+			d.Log.Debugf("Running docker command: %s %s", d.Docker.DockerCommand, strings.Join(args, " "))
+			err = d.Docker.Run(ctx, args, nil, writer, writer)
+			if err != nil {
+				return err
+			}
+		}
+
+		containerPasswdFileIn, err = os.Open(containerPasswdFileIn.Name())
+		if err != nil {
+			return err
+		}
+
+		// Update /etc/passwd and /etc/group with the new user UID and GID
+		scanner := bufio.NewScanner(containerPasswdFileIn)
+		containerUid := ""
+		containerGid := ""
+		containerHome := ""
+
+		re := regexp.MustCompile(fmt.Sprintf(`^%s:(?P<password>x?):(?P<uid>.*):(?P<gid>.*):(?P<gcos>.*):(?P<home>.*):(?P<shell>.*)$`, containerUser))
```

Comment:
why are we parsing the `/etc/group` and `/etc/passwd` file and rewriting it? What prevents us from simply copying the file over?

Reply:
I did not find a way to write in place in a file in go. You know how to do it?

Reply:
apologies I misunderstood the code. We are mutating the file, since we override `containerUser` based on the regex failing. So I think the implementation is sound